Make docstrings-style compliant and add assert messaging

zm711 opened the pull request:
So I switched most of the docstrings over to the SI format (I'm sure I'm missed some though) and I tried to add assert messaging to help users.

zm711 commented:
The tests I broke were designed to prevent users from messing up. I'll take a look.

zm711 commented:
So one of the test failures was because I chose the type of error from `Value` to `Type`. I think it is a `TypeError` so I just changed that test. 

The other set of failures were due to the fixture not specifying the `contact_shape_params`. Since the shape params are specific for the contact shape I think having an assert to prevent a user from doing this wrong is actually important, so I added the correct shape parameters for the fixture, but if you want me to revert that fixture and delete the assert confirming the correct mix of contact shape and contact shape params I could do that as well. 

Coverage drop is just because I made some asserts multiple lines and not all asserts are being tested.

zm711 commented on src/probeinterface/generator.py:
```diff
@@ -178,6 +192,10 @@ def generate_linear_probe(
     """
 
+    assert (
+        _default_dict[contact_shapes] in contact_shape_params.keys()
```
```suggestion
        _default_shape_to_params[contact_shapes] in contact_shape_params.keys()
```

zm711 commented:
Okay I'm all done with everything -- I won't push anything unless I get feedback. 

As a reminder (to check super carefully):
the assert I added in the generator (do you want me to delete and revert the test or do you think it is a good idea to prevent users from specifying the wrong shape params--also not sure if there is a specific spikeinterface test that could be affected by this change).

alejoe91 commented:
Thanks @zm711 

I'll review between today and tomorrow!

alejoe91 commented on src/probeinterface/utils.py:
```diff
@@ -47,7 +47,7 @@ def import_safely(module: str) -> ModuleType:
 
 
-def combine_probes(probes: Probe, connect_shape: bool = True) -> Probe:
+def combine_probes(probes: list[Probe], connect_shape: bool = True) -> Probe:
```
@zm711 in general, I believe that `list` is supported from Python3.9, otherwise we need `from __future__ import annotations`.

Since 3.8 will be alive until end of next year, let's be compliant with that.

zm711 replied:
Done.

alejoe91 reviewed (changes requested):
Thanks @zm711 

I requested a small change regarding the use of `|` and `list[]` for typing, which I believe needs `from __future__ import annotations`.

Could you add that when needed?

alejoe91 commented on src/probeinterface/generator.py:
```diff
@@ -99,26 +104,26 @@ def generate_multi_columns_probe(
     ypitch: float = 20,
     y_shift_per_column: Optional[np.array | list] = None,
-    contact_shapes: str = "circle",
+    contact_shapes: "circle" | "rect" | "square" = "circle",
```
The use of  `|` still requires `from __future import annotations` for Python3.8

zm711 replied:
This one already has `from __future__ import annotations` :)

zm711 commented:
Just to be safe I added future to all files (like Heberto has said, it's easy to miss that since we only test on 3.10), so since ProbeInterface is a pretty small code base I think it safer to just have from future everywhere and then we can delete it later.

I also added .ds_store to the gitignore since I do some work on Macs and don't want to accidentally upload a bunch of folder organization files :)

coverage failed which caused the test to fail. Could you try just re-running the test :)
